bears/general/LineBreakBear.py

```diff
@@ -0,0 +1,135 @@
+from queue import Queue
+
+from coalib.results.AbsolutePosition import AbsolutePosition
+from coalib.results.Diff import Diff
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.Result import Result, RESULT_SEVERITY
+from coalib.results.SourceRange import SourceRange
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+
+from bears.general.AnnotationBear import AnnotationBear
+from bears.general.GenericHelpers import get_specified_block_range
+from bears.general.LineLengthBear import LineLengthBear
+
+
+class LineBreakBear(LocalBear):
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_FIX = {'Formatting'}
+
+    def run(self,
+            filename,
+            file,
+            language: str,
+            dependency_results,
+            use_spaces=True,
+            indentation_width: int=SpacingHelper.DEFAULT_TAB_WIDTH,
+            max_line_length: int=79,
+            coalang_dir: str=None):
+        """
+        Runs the LineLengthBear and suggests alternate solutions where lines
+        exceed the max_line_length.
+
+        :param filename:
+            Name of the file that needs to be checked.
+        :param file:
+            File that needs to be checked in the form of a list of strings.
+        :param language:
+            Language of the file it is running on.
+        :param dependency_results:
+            Results given by the AnnotationBear and the LineLengthBear.
+        :param use_spaces:
+            Inserts spaces instead of tabs for indentation.
+        :param tab_width:
+            No. of spaces to add before the newly created line.
+        :param max_line_length:
+            Maximum number of characters for a line, the newline character
+            being excluded.
+        :param dependency_results:
+            Results given by the AnnotationBear.
+        :param coalang_dir:
+            Full path of external directory containing the coalang
+            file for language.
+        """
+        lang_settings_dict = LanguageDefinition(
```

Comment:
why call this a dict if it isn't really a dict? -> gets especially confusing in line 59 ;)